File: MyApp.API/Models/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using MyApp.API.Common;

namespace MyApp.API.Models
{
    public class Order
    {
        public int Id { get; set; }
        public DateTime DateCreated { get; set; }
        public DateTime DateUpdated { get; set; }
        public OrderStatus Status { get; set; }
        public double TotalAmount { get; set; }
        public ICollection<OrderItem> OrderItems { get; set; }
        public User User { get; set; }
        public int UserId { get; set; }
    }
}

File: MyApp-SPA/src/app/_models/order.ts
import { OrderStatus } from './enums';
import { Product } from './product';

export interface Order {
    id: number;
    dateCreated: Date;
    dateUpdated: Date;
    status: OrderStatus;
    totalAmount: number;
}

export interface OrderDetails {
    id: number;
    dateCreated: Date;
    dateUpdated: Date;
    status: OrderStatus;
    totalAmount: number;
    orderItems: OrderItem[];
}

export interface OrderItem {
    id: number;
    quantity: number;
    product: Product;
}
